x-pack/plugins/maps/common/descriptor_types/data_request_descriptor_types.ts

```diff
@@ -63,6 +63,20 @@ export type DataMeta = Partial<VectorSourceRequestMeta> &
   Partial<VectorStyleRequestMeta> &
   Partial<ESSearchSourceResponseMeta>;
 
+export type StyleMetaData = {
+  // key is field name for field requiring style meta
+  [key: string]:
+    | {
```

Comment:
Do you mind extracting these definitions to a named type? 

That way, any future TS-errors will be more readable, saying something like 

> foo is not compatible with `ExtendedStatsStyleMetaData`